### ISSUE TITLE
Allow support for disabling individual languages

### ISSUE THREAD

**Issue by mindriot101** [sheerun/vim-polyglot#50]
For example opening a large csv file with the csv plugin is extremely slow. I think enabling disabling the loading of the csv plugin with a setting in the vimrc file would be useful. It could be added during the dynamic loading section perhaps?

**Comment by dirtyrottenscoundrel** [comment]
I would also find an exclude list quite useful. Your project significantly cut down the number of plugins I had to manage manually, but occasionally, I'd like to try a different plugin for a language. For example, the included vim-protobuf plugin is quite out of date. I found [a new one](https://github.com/belltoy/vim-protobuf) that I'd like to test, but I don't see a simple way to avoid conflicts with the included version. I could fork this project, update the source list, and build a new version, but that's fairly heavy to run a quick test.

**Comment by jasonmobley** [comment]
+1

(for basically the same reason given by @dirtyrottenscoundrel)

**Comment by sheerun** [comment]
I temperately have little spare time, so a PR is welcome!

**Comment by dirtyrottenscoundrel** [comment]
I'm willing to take a crack at it if you can point me in the general direction @sheerun .

It looks like you're concatenating individual files from `ftdetect` into `ftdetect/polyglot.vim` during the build process. Would it be possible to wrap each appended file in a conditional check against a list of disabled plugins? Feels a little ugly, but I didn't see any other clear place to short-circuit the loading process.

Something like:

```viml
" ~/.vimrc
let g:polyglot_disabled = ['protobuf']
```

```viml
" ftdetect/polyglot.vim
if index(g:polyglot_disabled, 'protobuf') != -1
  autocmd BufNewFile,BufRead *.proto setfiletype proto
endif
```

**Comment by sheerun** [comment]
@dirtyrottenscoundrel Yup. I'm also thinking about something like following on top of each file:

```
if exists('g:polyglot_disabled') and index(g:polyglot_disabled, 'protobuf') != -1
  finish
endif
```

**Comment by tvjg** [comment]
Finally got around to making [a branch](https://github.com/tvjg/vim-polyglot/tree/allow-disabling). Seems to work nicely and the tests are passing, but I've accidentally introduced a non-obvious bug.

As discussed, a [guard](https://github.com/tvjg/vim-polyglot/blob/allow-disabling/plugin_guard.vim.template) is [prepended](https://github.com/tvjg/vim-polyglot/blob/allow-disabling/build#L62) to the top of each downloaded file. This occurs before the language packs are copied. Then, as files are copied, any identically named files [wind up being joined](https://github.com/sheerun/vim-polyglot/blob/master/build#L62). For example, `vim-coffee-script` and `vim-css-color` [both](https://github.com/ap/vim-css-color/blob/master/after/syntax/html.vim) [include](https://github.com/kchmck/vim-coffee-script/blob/master/after/syntax/html.vim) `after/syntax/html.vim`. This leads to [an undesirable result](https://github.com/tvjg/vim-polyglot/blob/allow-disabling/after/syntax/html.vim#L44). We get two overlapping guards. In effect, if `coffee-script` is added to `g:polyglot_disabled`, some bit of `css-color` will be unintentionally disabled as well.

Any thoughts @sheerun ?

**Comment by tvjg** [comment]
I think I got it now. I went back to the original idea of wrapping the file contents instead of exiting the file early. Opened a PR.

**Comment by sheerun** [comment]
Thank you for PR! It is now implemented (and will be released soon in v1.0.0)
